File: skypiea-client/src/main/java/com/skypiea/client/service/LoginService.java
package com.skypiea.client.service;

import com.skypiea.common.result.SPResult;

/**
 * 作者: huangwenjian
 * 描述:
 * 创建时间: 2017-03-23 09:49
 */
public interface LoginService {

    SPResult login(String username, String password);

    SPResult loginByShiro(String username, String password);

    //根据token取用户信息
    SPResult getUserByToken(String token);

    //安全退出
    SPResult logout(String token);

    SPResult logoutByShiro(String token);
}

File: skypiea-client/src/main/resources/templates/script/service/loginService.js
(function (angular) {
    'use strict';

    angular.module('app').service('loginService', ['$http', function ($http) {

        this.login = function (username, password, fn) {
            $http({
                url: '/client/user/loginByShiro',
                method: 'POST',
                params: {
                    username: username,
                    password: password
                }
            }).then(function (response) {
                fn(response);
            })
        }

        this.logout = function (accessToken, fn) {
            $http({
                url: '/client/user/logoutByShiro/' + accessToken,
                method: 'GET'
            }).then(function (response) {
                fn(response);
            });
        }

    }]);

})(angular)